### PR TITLE
[NFC] Remove explicit bitcode enumeration from BitCodeFormat.rst

### PR DESCRIPTION
**Pull request by cjappl**
As discussed in #102555 , this explicit listing of the bitcodes is out of date, and had fallen out of date in the past as well.

This PR deletes the explicit listing, and points users to where they can find the info if they need it.

### CODE REVIEW

**Comment by cjappl** [comment]
@nikic and @SLTozer for review, feel free to add any others! Totally open to different wording, any other feedback.

File: llvm/docs/BitCodeFormat.rst
Comment (SLTozer) [inline llvm/docs/BitCodeFormat.rst:1009]:
```suggestion
The integer codes are mapped to attributes as described in the
```
While we're looking at this, what does "well-known" mean in this context?

**Review by SLTozer — COMMENTED** [review]
New documentation LGTM, but wait for more comments in case someone has strong feelings about this.

File: llvm/docs/BitCodeFormat.rst
Comment (cjappl) [reply llvm/docs/BitCodeFormat.rst:1009]: Excellent question, I was just preserving the existing adjective but I agree it's unclear. I'll defer to someone with better knowledge of the area, I am happy to remove it if there is agreement.

**Comment by cjappl** [comment]
(just got official commit access, so adding reviewers officially, sorry for the "double ping")

File: llvm/docs/BitCodeFormat.rst
Comment (nikic) [reply llvm/docs/BitCodeFormat.rst:1009]: I guess that this is trying to distinguish pre-defined enum attributes from string attributes. Agree that just omitting the word would be clearer.

File: llvm/docs/BitCodeFormat.rst
Comment (nikic) [inline llvm/docs/BitCodeFormat.rst:1010]: I'd say `the AttributeKindCodes enumeration` here.

File: llvm/docs/BitCodeFormat.rst
Comment (nikic) [inline llvm/docs/BitCodeFormat.rst:1030]: The bitcode constant names and the C++ API names aren't really coupled, but I guess they do line of a bit better than the IR attribute names.

Comment (cjappl) [reply llvm/docs/BitCodeFormat.rst:1030]: Not sure if you wanted a change here or not, left it be for now.

File: llvm/docs/BitCodeFormat.rst
Comment (cjappl) [reply llvm/docs/BitCodeFormat.rst]: Fixed in latest

**Comment by cjappl** [comment]
Weekly ping of reviewers: @nikic and @ramosian-glider 

Status: all outstanding comments addressed

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**
